src/pages/HomePage/index.js

```diff
@@ -1,174 +1,133 @@
-    maxItemLengthFixed: SLIDER_IMG_LENGTH,
-    listLength: null,
-    scrollYPosition: 0
-  };
-
-  componentDidMount() {
-    this.timeoutSlider();
-    const objPerson = this.randomNumb(0, persons.length - 1);
-    const list = [];
-    const { mainPhoto, profilePhoto, pressPhoto } = objPerson;
-    list.push(mainPhoto, profilePhoto, pressPhoto);
-    this.setState({
-      defaultPerson: persons[objPerson],
-      listLength: list.length
-    });
+const HomePage = ({ people, articles, history }) => {
+  const [listIsOpen, setListIsOpen] = useState(false);
+  if (isEmpty(people) || isEmpty(articles)) {
+    return <Loading />;
   }
-
-  render() {
-    const { history } = this.props;
-    const { numberPhoto, currentIndex, maxItemLengthChanges } = this.state;
-    let newArticles = articles.slice(0, 7);
-    const {
-      id,
-      firstName,
-      lastName,
-      shortDescription,
-      mainPhoto,
-      profilePhoto,
-      pressPhoto
-    } = this.state.defaultPerson;
-    const list = [];
-    list.push(mainPhoto, profilePhoto, pressPhoto);
-    const filterData = persons.slice(0, maxItemLengthChanges);
-
-    return (
-      <div className="home-page">
-        <div className="home-page__client">
-          <div className="home-page__social">
-            <ButtonsSocial component="home-page" />
-            <div className="home-page__first-divider" />
-            <div className="home-page__number-slide">{numberPhoto}</div>
-          </div>
-          <div className="home-page__slider">
-            <img
-              src={list[currentIndex]}
-              alt="person photo"
-              className="home-page__slider-img"
-            />
-            <div className="home-page__info">
-              <h2 className="home-page__title">
-                {firstName} <br /> {lastName}
-              </h2>
-              <p className="home-page__type-person">{shortDescription}</p>
-            </div>
-            <div
-              onClick={() => history.push(`/profile/${id}`)}
-              className="home-page__link"
-            >
-              <p className="home-page__link-name">view profile</p>
-              <div className="home-page__link-svg">
-                <TriangleRight />
-              </div>
-            </div>
-          </div>
-          <div className="home-page__btn-img">
-            <ButtonList
-              currentIndex={currentIndex}
-              list={list}
-              stateNumb={this.stateNumb}
-            />
-          </div>
+  const indexRandomPerson = randomNumb(0, people.length - 1);
+  const person = people[indexRandomPerson];
+  let newArticles = articles.slice(0, 7);
+  const filterData = listIsOpen ? people : people.slice(0, SLIDER_IMG_LENGTH);
+
+  return (
+    <div className="home-page">
+      <PersonPhotoSlider {...person} history={history} />
+      <h3 className="home-page__client-title">Our clients</h3>
+      <SliderWrapper filterData={filterData} position={0} onClick={null} />
```

Comment:
onClick={null} ???
maybe onClick={()=>{}} if you don't need to send anythink in onClick